ocaml/typing/solver.ml

```diff
@@ -716,37 +727,57 @@ module Solver_mono (C : Lattices_mono) = struct
       generalize_topology ~log dst ~current_level ~generic_level u;
       update_level_v ~log dst generic_level u
 
-  type varcond = { b : 'a. 'a var -> bool; } [@@unboxed]
+  let copy_exists : type a. ?copy_to_level: int -> a C.obj -> a var -> (a var) option =
+    fun ?copy_to_level obj v ->
+      Option.bind copy_to_level (fun level ->
+        match v.vlower, v.vupper with
+        | Amorphvar (u1, f) :: _, Amorphvar (u2, _) :: _ ->
```

Comment:
This also needs to check that the morphisms are both the identity morphism. Possibly we need to ask for an `is_identity` function on morphisms to do this if we don't already.